src/main/java/org/fiware/iam/ServiceMapper.java

```diff
@@ -1,103 +1,130 @@
-						.setEndpoint(endpoint)
-						.setType(EndpointType.TRUSTED_PARTICIPANTS))
-				.toList();
-	}
-
-	/**
-	 * Map a list of string-entries, encoding TrustedIssuers endpoints to a list of {@link EndpointEntry} with
-	 * type {{@link EndpointType.TRUSTED_ISSUERS}
-	 */
-	default List<EndpointEntry> issuersToEntries(List<String> endpoints) {
-		if (endpoints == null) {
-			return null;
-		}
-		return endpoints.stream()
-				.map(endpoint -> new EndpointEntry()
-						.setEndpoint(endpoint)
-						.setType(EndpointType.TRUSTED_ISSUERS))
-				.toList();
-	}
-
-	/**
-	 * Return issuer endpoints from the {@link EndpointEntry} list to a list of strings
-	 */
-	default List<String> entriesToIssuers(List<EndpointEntry> endpoints) {
-		if (endpoints == null) {
-			return null;
-		}
-		return endpoints.stream()
-				.filter(entry -> entry.getType().equals(EndpointType.TRUSTED_ISSUERS))
-				.map(EndpointEntry::getEndpoint)
-				.toList();
-	}
-
-	/**
-	 * Return participant endpoints from the {@link EndpointEntry} list to a list of strings
-	 */
-	default List<String> entriesToParticipants(List<EndpointEntry> endpoints) {
-		if (endpoints == null) {
-			return null;
-		}
-		return endpoints.stream()
-				.filter(entry -> entry.getType().equals(EndpointType.TRUSTED_PARTICIPANTS))
-				.map(EndpointEntry::getEndpoint)
-				.toList();
-	}
-
-}
+    Service map(ServiceVO serviceVO);
+
+
+    default Map<String,Collection<Credential>> map(Map<String,ServiceScopesEntryVO> value){
+        return Optional.ofNullable(value)
+                .orElseGet(Map::of)
+                .entrySet()
+                .stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().stream().map(this::map).toList()));
+    }
+
+    default Map<String,ServiceScopesEntryVO> mapCredentials(Map<String,Collection<Credential>> value){
+        Map<String,ServiceScopesEntryVO> answer = new HashMap<>();
+        Optional.ofNullable(value)
+                .orElseGet(Map::of)
+                .entrySet().forEach(entry -> {
+            ServiceScopesEntryVO credentialVOS = new ServiceScopesEntryVO();
+            entry.getValue().stream().map(this::map).forEach(credentialVOS::add);
+            answer.put(entry.getKey(), credentialVOS);
+        });
+        return answer;
+    }
+
+    ServiceVO map(Service service);
+
+    default Credential map(CredentialVO credentialVO) {
+        if (credentialVO == null) {
+            return null;
+        }
+        Credential credential = new Credential()
+                .setCredentialType(credentialVO.getType());
+        List<EndpointEntry> trustedList = new ArrayList<>();
+        Optional.ofNullable(issuersToEntries(credentialVO.getTrustedIssuersLists())).ifPresent(trustedList::addAll);
+        Optional.ofNullable(participantsToEntries(credentialVO.getTrustedParticipantsLists())).ifPresent(trustedList::addAll);
+        credential.setTrustedLists(trustedList);
+        return credential;
+    }
+
```

Comment:
isn't that a mapping that mapstruct can construct?

Reply:
Hem for the trustedLists field, two other fields are mapped and then merged into one. At least I didn't try something like that yet